EKF/ekf_helper.cpp

```diff
@@ -509,14 +509,14 @@ bool Ekf::realignYawGPS()
 			_state_reset_status.quat_change = quat_before_reset.inversed() * _state.quat_nominal;
 
 			// add the reset amount to the output observer buffered data
-			// Note q1 *= q2 is equivalent to q1 = q2 * q1
+			// Note q1 *= q2 is equivalent to q1 = q1 * q2
```

Comment:
This correct but confusing given the use of the Eigen library. I propose we remove this comment line and change line 514 to:

_output_buffer[i].quat_nominal = _state_reset_status.quat_change * _output_buffer[i].quat_nominal;

to make the multiplication order more obvious.

Reply:
Have checked the library and the self multiplication operator does reverse the order compared to the eigen convention which is not good. I recommend we discontinue use of the self multiplication operator to avoid confusion and replace it with_output_buffer[i].quat_nominal = _output_buffer[i].quat_nominal * _state_reset_status.quat_change;

Reply:
It might be best to change line 509 so that the delta quaternion it produces is from old -> new and can be applied using an order that is consistent with use of the quaternion product in other places when an incremental rotation is applied, eg:

_state_reset_status.quat_change =  * _state.quat_nominal * quat_before_reset.inversed();